sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncAccountRepository.kt

```diff
@@ -288,6 +300,7 @@ class AppSyncAccountRepository @Inject constructor(
 
         return when (val result = syncApi.getDevices(token)) {
             is Error -> {
+                connectedDevicesCached.clear()
```

Comment:
why is this added? if there is a blip in the network and `getDevices` call fails, do we want to clear the cached list?

Reply:
It's a convenient, quick solution, for us to have the latest response from get device in memory, so we can use it in the prompt logic. It's fine to clear on error, and only update/store on success.